ui/pages/confirmations/hooks/useLedgerConnection.ts

```diff
@@ -22,15 +23,18 @@ import { useConfirmContext } from '../context/confirm';
 
 const useLedgerConnection = () => {
   const dispatch = useDispatch();
-  const { currentConfirmation } = useConfirmContext<SignatureRequestType>();
+  const { currentConfirmation } = useConfirmContext<
+    SignatureRequestType & TransactionMeta
+  >();
   const ledgerTransportType = useSelector(getLedgerTransportType);
   const transportStatus = useSelector(getLedgerTransportStatus);
   const webHidConnectedStatus = useSelector(getLedgerWebHidConnectedStatus);
 
   let from: string | undefined;
-  // todo: extend to other confirmation types
-  if (currentConfirmation?.msgParams) {
-    from = currentConfirmation.msgParams.from;
+  const confirmationParams =
+    currentConfirmation?.msgParams || currentConfirmation?.txParams;
+  if (confirmationParams) {
```

Comment:
Minor, could we avoid the `let` and this `if` and just use two optional chains such as:

`currentConfirmation?.msgParams?.from ?? currentConfirmation?.txParams?.from`

Reply:
Done 👍